pkg/minikube/node/cache.go

```diff
@@ -43,7 +43,7 @@ const (
 
 // BeginCacheKubernetesImages caches images required for kubernetes version in the background
 func beginCacheKubernetesImages(g *errgroup.Group, imageRepository string, k8sVersion string, cRuntime string) {
-	if download.PreloadExists(k8sVersion, cRuntime) {
+	if imageRepository == "" && download.PreloadExists(k8sVersion, cRuntime) {
```

Comment:
Add:

TODO: remove imageRepository check once https://github.com/kubernetes/minikube/issues/7695 is fixed.

Reply:
Thanks for the review. Added a TODO note.